docs/operator-dependencies-and-requirements.md

```diff
@@ -0,0 +1,64 @@
+# Operator Dependency and Requirement Resolution
+
+## Overview on Dependency Principles
+
+OLM manages the dependency resolution and upgrade lifecycle of running operators.
+
+An operator's dependencies are defined through the (Group, Version, Kind) of CustomResourceDefinitions (CRDs). An operator cannot define a dependency on a specific version (e.g. `etcd-operator v0.9.0`) or instance (e.g. `etcd v3.2.1`) of an operator. This encourages operator authors to depend on the interface and not the implementation; thereby allowing operators to update on individual release cycles.
+
+OLM manages operator dependency resolution very similarily to OS package managers, such as apt/dkpg and yum/rpm. However, given that operators are always running, OLM attempts to ensure that it never deploys a combination of operators that do not work with each other.
+
+This means that OLM will never:
+
+* install a set of operators that require APIs that can't be provided
+* update an operator in a way that breaks another that depends upon it
+
+### Dependency Resolution Alogorithm
```

Comment:
I think that this is a bit too low-level. Can we describe how dependency resolution works without psuedocode like this?

Reply:
Will do.